translator/GCP_cloud_function/cloud_function/icone_translator.py

```diff
@@ -136,33 +106,30 @@ def get_road_direction(coordinates):
     else:
         direction = 'southbound'
 
-    if lat_dif ==0 and long_dif==0 :
-        direction= None
+    if lat_dif == 0 and long_dif == 0:
+        direction = None
 
     return direction
 
+
 def parse_direction_from_street_name(street):
-    #function to parse direction from street name
-    if not street :
-        return  None
-    street_char=street[-1]
-    if street_char == 'N':
-        direction='northbound'
-    elif street_char =='S':
-        direction ='southbound'
-    elif street_char =='W':
-        direction ='westbound'
-    elif street_char =='E':
-        direction ='eastbound'
+    # function to parse direction from street name
+    street_char = street[-1]
```

Comment:
Do we still need the check in here for empty street? What happens when you pass this function an empty string?

Reply:
we added a case if street is empty. In practice this will never happen because this function will never be called if street is empty.

Reply:
Ah, now I see the road_name check above. We can make that code a bit clearer on line 279 - 290 by reusing the "road_name" variable and changing as follows:
```
# road_name
    road_name = incident['location'].get('street', '')
    if not road_name:
        return None
    properties['road_name'] = road_name

    # direction
    direction = parse_direction_from_street_name(road_name)
    if not direction:
        direction = get_road_direction(geometry['coordinates'])
        if not direction:
            return None
```

Reply:
I just made  your requested change.